stairway/src/test/java/bio/terra/stairway/FlightMapTest.java

```diff
@@ -153,4 +188,48 @@ public void serDesExceptions() {
     Assertions.assertThrows(
         JsonConversionException.class, () -> FlightMap.create(new ArrayList<>(), "garbage"));
   }
+
+  @Test
+  public void guavaTypes() {
+    FlightMap flightMap = new FlightMap();
+
+    final String key = "immutableList";
+    ImmutableList<Integer> immutableListIn = ImmutableList.of(1, 2, 3, 4);
+    flightMap.put(key, immutableListIn);
+
+    ImmutableList<Integer> immutableListOut = flightMap.get(key, ImmutableList.class);
+    Assertions.assertEquals(immutableListIn, immutableListOut);
+  }
+
+  @Test
+  public void typeRef() {
+    FlightMap flightMap = new FlightMap();
+    final String key = "key";
+
+    Map<MyEnum, FlightsTestPojo> testMapIn = new HashMap<>();
+    testMapIn.put(MyEnum.FOO, new FlightsTestPojo().anint(1).astring("first"));
+    testMapIn.put(MyEnum.BAR, new FlightsTestPojo().anint(2).astring("second"));
+    testMapIn.put(MyEnum.BAZ, new FlightsTestPojo().anint(3).astring("third"));
+    flightMap.put(key, testMapIn);
```

Comment:
Does `Map.of` not work here? That would be a slightly easier way to write this.

Reply:
Weird, when I replace the repeated push with `Map.of`:
```
{"FOO":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"first","anint":1}],"BAZ":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"third","anint":3}],"BAR":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"second","anint":2}]}
```
I get different JSON that doesn't play nice with either deserializer:
```
{"FOO":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"first","anint":1}],"BAZ":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"third","anint":3}],"BAR":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"second","anint":2}]}
```
With the code as-is I get:
```
["java.util.HashMap",{"BAR":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"second","anint":2}],"FOO":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"first","anint":1}],"BAZ":["bio.terra.stairway.fixtures.FlightsTestPojo",{"astring":"third","anint":3}]}]
```
Given the `TypeReference<T>` is working for the WSM case, I'm inclined not to dig too deep...

Reply:
That's unfortunate. I would much rather see support of JDK types like `Map.of()` than guava types in stairway, although I don't know what's needed to fix this. TDR may end up using `Map.of()` in the future so I guess we'll figure it out then.